osu.Game/Screens/Play/MasterGameplayClockContainer.cs

```diff
@@ -49,33 +49,38 @@ public class MasterGameplayClockContainer : GameplayClockContainer
         private readonly BindableDouble pauseFreqAdjust = new BindableDouble(1);
```

Comment:
It's both funny and scary how something like this can cause tests to fail in a brain-melting way (definitely worth adding test coverage for):
```suggestion
        private readonly BindableDouble pauseFreqAdjust = new BindableDouble();
```
(`GameplayClockContainer` is now paused by default, so the adjustment should be `0`, otherwise will be stuck to `1` until `IsPaused` is toggled off and back on)

To explain why this is only failing when multiple test cases run in the same session (and I'm gonna pick the `TestSeekPerformsInGameplayTime(0.5d, -200.0d, true)` test case as an example while explaining): Since `pauseFreqAdjust` is set to 1 by default (which is incorrect), the first `ProcessFrame` call that the offset clocks receive are when `Reset` is called in the test case:
https://github.com/ppy/osu/blob/7cb7b03cee678db55530d8134d39295089a6a216/osu.Game.Tests/Gameplay/TestSceneMasterGameplayClockContainer.cs#L86-L88

And, since another test has ran in the same session, `OsuSetting.AudioOffset` will already be populated with a certain offset (let's say -200ms) at the point of calling `Reset`, which ends up making `userGlobalOffsetClock.RateAdjustedOffset` matching that offset at the *current* rate (which is 1, not 0.5, since the track is recreated per-run, and the rate adjustment is applied in a later step).

Then, since on this test case the clock remains stopped, the second `ProcessFrame` call only happens during the `Seek(2500)` step, and the failure occurs there, since the following happens when `Seek` gets called:
 - The *current* `totalAppliedOffset` (which is going to be -200ms from above) is subtracted from the proposed time (2500ms), so the base seek call receives 2700ms
 - Seeking is performend and `GameplayClock.UnderlyingClock.ProcessFrame()` is called.
 - The `totalAppliedOffset` *changes* from -200ms to -100ms since this is the first time `ProcessFrame()` gets called while the track rate is actually 0.5 instead of 1.

This ends up making a difference in `CurrentTime` from the expected one since -200ms was subtracted from it at first but now `totalAppliedOffset` has been corrected to be -100ms instead (so it ends up with 2600ms instead of the proposed 2500ms).

---

With `pauseFreqAdjust` being changed to be 0 by default, `RateAdjustedOffset` will never be touched and tests will pass just fine.

While it may seem awkward that `HardwareCorrectionOffsetClock` only gets processed during seeks from the explaination above, this is only happening because of this edge case (`IsPaused = true && pauseFreqAdjust = 1`). If `IsPaused = false && pauseFreqAdjust = 1`, offsets will be updated per-frame, and if `IsPaused = true && pauseFreqAdjust = 0`, offsets will never be updated neither per-frame nor in seeks.

The only potential issue is when offsets are changed *and* the pause state changed without any prior frames processing the offset change, but I'm not sure that can be solveable anyways (and I might as well be proven wrong on this statement).

Reply:
What an ordeal. Thanks for the investigation. Will see what I can do with this knowledge.

Reply:
Applying your proposed change breaks other tests, so will need a bit of further thought:

![JetBrains Rider-EAP 2022-03-24 at 05 34 15](https://user-images.githubusercontent.com/191335/159849323-cf86a884-52c4-4199-a267-9bf7f2db0d9f.png)

Reply:
I've fixed the remaining tests and added some test coverage for this scenario. I'm still not 100% on this, but it's something to be considered at a later point. Specifically, I'm not sure how I feel about the gameplay clock potentially being in an incorrect state for the "current" frame because it's being updated as a child of `Player`.

At some point we will want to do an intensive consistency pass on the game loop in general to ensure we aren't inducing any single-frame delays as a result of having components processed too late.